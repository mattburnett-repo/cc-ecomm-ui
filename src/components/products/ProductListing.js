import BrowseSearch from "../nav/BrowseSearch";

export default function ProductListing () {
    return (
        <div>
            <div role='presentation' aria-label="product-category">product category goes here</div>
            <BrowseSearch />

            <div role="presentation" aria-label="product-listing">
                <div>product 01</div>
                <div>product 02</div>
                <div>product 03</div>
            </div>
        </div>
    )
}