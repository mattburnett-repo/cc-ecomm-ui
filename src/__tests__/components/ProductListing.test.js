// 20211217: Wireframe https://wireframe.cc/D5VIOi

import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import ProductListing from '../../components/products/ProductListing';

describe('<ProductListing /> component tests', () => {
    beforeEach(() => render(<ProductListing />));

    it('should display the selected product category', () => {
        screen.getByRole('presentation', {name: /product-category/i});
    });
    it('should show browse and search', () => {
        screen.getByRole('presentation', {name: /browse-search/i});
    });
    it('should show individual product listings', () => {
        screen.getByRole('presentation', {name: /product-listing/i});
    });
}); // end component

describe('<ProductListing /> feature tests', () => {
    beforeEach(() => render(<ProductListing />));

    it('displays individual product listing', () => {
        expect(true).toBeFalsy();
    })

    it('creates a snapshot', () => {
        const tree = renderer.create(<ProductListing />).toJSON();
        expect(tree).toMatchSnapshot();  
    })
}); // end feature

describe('<ProductListing /> unit / functional tests', () => {
    beforeEach(() => render(<ProductListing />));

    it('selects a product category', () => {
        expect(true).toBeFalsy();
    });
    it('accepts search text', () => {
        const theElement = screen.getByRole('textbox', {name: /search-terms/i});
        userEvent.type(theElement, 'here is some search text');
        
        expect(theElement).toHaveValue('here is some search text');
    });
    it('clicks the search button', () => {
        const theVal = screen.getByRole('button', {name: /search/i});
        fireEvent.click(theVal);
    });
}); // end unit / functional