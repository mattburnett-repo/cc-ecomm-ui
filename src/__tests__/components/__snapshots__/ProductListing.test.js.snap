// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductListing /> feature tests creates a snapshot 1`] = `
<div>
  <div
    aria-label="product-category"
    role="presentation"
  >
    product category goes here
  </div>
  <div
    aria-label="browse-search"
    role="presentation"
  >
    <div
      aria-label="product-categories"
      role="presentation"
    >
      select product categories
    </div>
    <input
      aria-label="search-terms"
      placeholder="Enter search terms"
    />
    <button
      aria-label="search-button"
    >
      Search
    </button>
  </div>
  <div
    aria-label="product-listing"
    role="presentation"
  >
    <div>
      product 01
    </div>
    <div>
      product 02
    </div>
    <div>
      product 03
    </div>
  </div>
</div>
`;
